apis/nucleus/src/components/listbox/ListBoxInline.jsx

```diff
@@ -71,7 +82,13 @@ export function ListBoxInline({ app, fieldIdentifier, stateName = '$', options =
   }
 
   const theme = useTheme();
-  const [model] = useSessionModel(listdef, app, fieldName, stateName);
+
+  let model;
+  if (sessionModel) {
+    model = sessionModel;
+  } else {
+    [model] = useSessionModel(listdef, app, fieldName, stateName);
+  }
```

Comment:
this breaks [Rules of Hooks](https://reactjs.org/docs/hooks-rules.html)
hooks should not be called conditionally
and will probably need a more complicated workaround

Reply:
I read up a bit and I found two typical workarounds:
- The first option is to *wrap* the conditional parts inside a component which is then executed on render (since this is a valid call)
- The second option is to always call the hook but then overwrite it conditionally

The first option is not viable from what I can see (we need this to happen *before* render). *Maybe* I can wrap the Listbox like this to acquire that behaviour:
```
<SessionModel> <-- Conditional logic wrapped in component
  <SelectionsAPI> <-- Conditional logic wrapped in component
    <ListBoxInline app={app} fieldIdentifier={fieldIdentifier} stateName={stateName} options={options} />,
  </SelectionsAPI>
</SessionModel>
```

The second option means unnecessary code will run on each hook call. However, assuming the hook does its job in cleaning up when unmounted (as hooks should do) I don't see any issue with this.

Thoughts?

Reply:
I think the solution you have in place now looks ok and not too over-engineered.

An alternative I've used in similar situations is to return different components based on whats available. Example to illustrate it in this context:

```js
const ListboxInLine = (props) => {
  // Listbox related logic
  return model ? <WithModel model={model} {...props} /> : <WithHook {...props} />;
};

const WithModel = ({model, ...props}) => {
  return <UI model={model} {...props} />;
}

const WithHook = ({...props}) => {
  const [model] = useSessionModel(...);
  return <UI model={model} {...props} />;
}

const UI = ({ model, ...props}) => {
  // Use `model` somehow 
  return (<Grid>...</Grid>);
}
```

Another approach is to wrap the `useSessionModel()` in your own hook which accepts the sessionModel along with the parameters to the existing hook. Might be a bit overkill in this case but it would make the code in the React component more concise:

```js
import useSessionModel from '....';

const useSessionModelWrapped = (sessionModel, listdef, app, fieldName, stateName) => {
  const hookSessionModel = useSessionModel(...);
  return sessionModel || hookSessionModel;
} 

const ListboxInline = () => {
  const model = useSessionModelWrapped(sessionModel, listdef, app, fieldName, stateName);
  // ...
}
```

Reply:
Yep, I just pushed that change now. For the `useObjectSelections` I need to send in an empty object to prevent an error but it has same effect. Good suggestions.